h2o-docs/src/product/data-munging/target-encoding.rst

```diff
@@ -146,136 +216,151 @@ Now we will perform target encoding on ``addr_state`` to see if this representat
+    # Set the predictor to be "addr_state"
+    predictor = "addr_state"
 
--  ``holdout_type``: "KFold"
--  ``blended_avg``: TRUE
--  ``noise_level``: NULL (by default it will add 0.01 \* range of y of random noise)
+    from h2o.targetencoder import TargetEncoder    
+    te_map = TargetEncoder(x=predictor, y=response, fold_column="fold", blended_avg= True, inflection_point = 3, smoothing = 1)
+    te_map.fit(train)
 
-.. code:: r
+Transform Target Encoding
+'''''''''''''''''''''''''
 
-    ext_train <- h2o.target_encode_apply(train, x = list("addr_state"), y = response, 
-                                         target_encode_map = te_map, holdout_type = "KFold",
-                                         fold_column = "fold",
-                                         blended_avg = TRUE, noise_level = 0, seed = 1234)
+Apply the target encoding to our training and testing data. 
 
-    head(ext_train[c("addr_state", "fold", "TargetEncode_addr_state")])
+**Apply Target Encoding to Training Dataset** 
 
-::
+For our training data, we will use the parameters:
 
-    ##   addr_state fold TargetEncode_addr_state
-    ## 1         AK    0               0.1212239
-    ## 2         AK    0               0.1212239
-    ## 3         AK    0               0.1212239
-    ## 4         AK    0               0.1212239
-    ## 5         AK    0               0.1212239
-    ## 6         AK    0               0.1212239
+-  ``holdout_type="kfold"``
+-  ``blended_avg=TRUE``
+-  ``noise=0.2``
 
-For our testing data, we will use the parameters:
+.. example-code::
+   .. code-block:: r
 
--  ``holdout_type``: "None"
--  ``blended_avg``: False
--  ``noise_level``: 0
+    ext_train <- h2o.target_encode_transform(train, x = list("addr_state"), y = response, 
+                                             target_encode_map = te_map, holdout_type = "kfold",
+                                             fold_column="fold", blended_avg = TRUE, 
+                                             inflection_point=3, smoothing=1, seed = 1234)
 
-We do not need to apply any of the overfitting prevention techniques since our target encoding map was created on the training data, not the testing data.
+   .. code-block:: python
 
-.. code:: r
+    encoded_train = te_map.transform(frame=train, holdout_type="kfold", seed=1234)
 
-    ext_test <- h2o.target_encode_apply(test, x = list("addr_state"), y = response,
-                                        target_encode_map = te_map, holdout_type = "None",
-                                        fold_column = "fold",
-                                        blended_avg = FALSE, noise_level = 0)
+**Apply Target Encoding to Testing Dataset**
 
-    head(ext_test[c("addr_state", "TargetEncode_addr_state")])
+We do not need to apply any of the overfitting prevention techniques because our target encoding map was created on the training data, not the testing data.
 
-::
+-  ``holdout_type="none"``
+-  ``blended_avg=FALSE``
```

Comment:
@angela0xdata blending is not about overfitting prevention. It is about imperfectness of our data. When we don't have enough data for a particular categorical level we want to use prior (means ignoring levels of the categorical variable) mean and blend it with what we know from posterior mean.

Reply:
@deil87 okay, I'll try this all again and let you know how this goes. But just an FYI, you're welcome to check out the branch and make edits directly in the file. Again, I'm not an expert. I was just trying to make the Python version work similarly to R.

Reply:
@angela0xdata That is actually what I'm doing :) Please take a look at changes that I'have pushed. I know that this task might be a painful one for you... I'm trying to help.

Reply:
Oh... yeah, I meant that I appreciated your updates, and in the future, you're always welcome to again :) ... sorry, that was pre-coffee.